src/app/sidebar/sidebar.tsx

```diff
@@ -0,0 +1,130 @@
+import classnames from "classnames";
+import React, { useState } from "react";
+import css from "./Sidebar.module.css";
+import {
+    Sidenav,
+    SidenavItems,
+    SidenavParent,
+    SidenavLink,
+    SidenavFilter,
+    OnChangeCallback,
+    OnChangeInput,
+} from "./sidenav";
+import { NavLink, useLocation, matchPath } from "react-router-dom";
+import styles from "./Sidebar.module.css";
+import i18n from "@dhis2/d2-i18n";
+import { HidePreventUnmount } from "../../components/utils";
+import { LinkItem, SidebarLinks, sidebarLinks } from "./sidebarLinks";
+
+interface SidebarNavLinkProps extends LinkItem {
+    disabled?: boolean;
+    end?: boolean;
+}
+
+const SidebarNavLink = ({ to, label, disabled, end }: SidebarNavLinkProps) => {
+    return (
+        <SidenavLink
+            to={to}
+            disabled={disabled}
+            LinkComponent={NavLink}
+            label={label}
+            end={end}
+        />
+    );
+};
+
+interface SidebarParentProps {
+    isFiltered: boolean;
+    label: string;
+    links: LinkItem[];
+}
+
+const SidebarParent = ({
+    label,
+    links = [],
+    isFiltered,
+}: SidebarParentProps) => {
+    const { pathname } = useLocation();
+
+    // Check if any of the children match the current path
+    // If they do, parent should be open by default
+    const routePathMatch = links.some((link) => {
+        return matchPath(link.to, pathname);
+    });
+
+    const forceOpen = (isFiltered && links.length > 0) || undefined;
+    return (
+        <SidenavParent
+            label={label}
+            initialOpen={routePathMatch}
+            forceOpen={forceOpen}
+        >
+            {links.map(({ to, label }) => (
+                <SidebarNavLink key={label} to={to} label={label} />
+            ))}
+        </SidenavParent>
+    );
+};
+
+export const Sidebar = ({ links = sidebarLinks }: { links: SidebarLinks }) => {
```

Comment:
I don't think the sidebar needs a `links` prop, we can use the sidebarLinks directly

Reply:
This was mostly done to make it easier to test.

Reply:
That could be done with mocking the sidebarLinks module as well, right? Right now I can't see a test for this though. But I suppose that you could just use the actual list for testing (which on the one hand shouldn't really have an impact on the test code itself and on the other hand actually ensures that tests break when you change the sidebar links when you change them, which enforces that to be a conscious decision)

Reply:
https://dhis2.atlassian.net/browse/DHIS2-15072